### DISCUSSION TITLE
Carousels work a little bit slow when "Low Power Mode" is on the iPhone

### DISCUSSION THREAD

**Discussion by Sergey98Am** [davidjerleke/embla-carousel#676] [Help]
https://github.com/davidjerleke/embla-carousel/assets/52923458/9fbc7f74-ea84-43ed-813a-1b707bc92d14

## Bug is related to

- [x] embla-carousel (core package)
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs (documentation)
- [ ] embla-carousel-docs (generator)

### Describe the bug

- Carousels work a little bit slow when "Low Power Mode" on iPhone. When comparing Embla to other carousels, they work smoother on lower mode than Embla. Is there an option to make performance quicker or is it a bug?
- Maybe the problem is not obvious in this video, if you try it yourself it will be

**Comment by davidjerleke** [comment]
@Sergey98Am what are you comparing? The other carousels you use for comparison, are they using CSS transitions or JavaScript transitions? Because Embla is using JavaScript transitions and some devices throttle `requestAnimationFrame` in some cases. [Low battery](https://popmotion.io/blog/20180104-when-ios-throttles-requestanimationframe/) is one of them and it seems like iOS devices does this quite generously.

**Reply by Sergey98Am** [reply]
Especially I compared with Splide and Swiper. Maybe yes, they use CSS transitions. Thank you anyway.

**Reply by davidjerleke** [reply]
@Sergey98Am I thinks CSS transitions are not throttled as much as JavaScript transitions with `requestAnimationFrame`. But on the other hand, there are other limitations with CSS transitions. I don’t know if there’s way around this.

**Reply by Sergey98Am** [reply]
Is it possible to add CSS mode on Embla like Swiper, or it's bad option? It's just interesting

**Reply by davidjerleke** [reply]
@Sergey98Am no, unfortunately it would seriously limit many of its features.

**Reply by Sergey98Am** [reply]
Ok, thank you for the quick replies

**Reply by davidjerleke** [reply]
@Sergey98Am I’ve been wanting to try switching to the browser native web Animation API but given the heavy work load with feature requests and questions it’s pretty far away in time I would say.

**Reply by Sergey98Am** [reply]
@davidjerleke Sorry to bother you, I have another question, which CSS transition limitations there are, can you say a few. I want to compare CSS and JS transition limitations. In the short way I want to know advantages of JS transition?

**Reply by davidjerleke** [reply]
@Sergey98Am,

With JS, you know the next exact position beforehand (because you store it a variable), before applying styles to the DOM which gives you full control in matters like creating a seamless loop effect without visual glitches, or always being responsive to user input like previous and next buttons. Many other libs lock the previous and next buttons for a while damaging accessibility because they need to hide their loop effect glitch that otherwise would be entirely visible.

With CSS, you don’t know the current carousel position beforehand and if you need that information you need to get computed styles from the DOM which triggers layout thrashing.

I think you can find more detailed information on the matter online using Google.

The web Animations API is the best of two worlds I guess, it gives the control of javascript with the performance of CSS. But as mentioned a refactor using it would be a big one that takes a lot of time and effort.

**Reply by Sergey98Am** [reply]
Thank you very much, this was very useful information for me. 👍🏻
